lib/pure/hashes.nim

```diff
@@ -136,6 +139,17 @@ proc hash*[T: Ordinal](x: T): Hash {.inline.} =
   ## Efficient hashing of other ordinal types (e.g. enums).
   result = ord(x)
 
+template multibyteHashImpl(result: Hash, x: typed, start, stop: int) =
+  var i = start
+  while i <= stop+1 - IntSize:
+    let n = cast[ptr int](unsafeAddr x[i])[]
```

Comment:
Can we have a fallback for hashing strings in the VM and a test.

Reply:
Also `while i < stop - IntSize:` is cleaner

Reply:
> Also `while i < stop - IntSize:` is cleaner

You almost had me there ;)

Your version is true only when `stop >= 9` or higher (length of `10`), mine is when `stop >= 7` (length of `8`).

Reply:
Argh off-by-one :/